engine/src/main/scala/cromwell/engine/workflow/lifecycle/WorkflowExecutionActor.scala

```diff
@@ -118,8 +211,131 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescript
       val configDescriptor = BackendConfigurationDescriptor("", ConfigFactory.load())
       val props = DummyBackendJobExecutionActor.props(jobDescriptor, configDescriptor)
       val key = jobDescriptor.key
-      context.actorOf(props, name = s"${jobDescriptor.descriptor.id}-BackendExecutionActor-${key.call.taskFqn}-${key.index}-${key.attempt}")
+      val actorName = s"${jobDescriptor.descriptor.id}-BackendExecutionActor-${key.call.fullyQualifiedName}-${key.index.getOrElse("")}-${key.attempt}"
+      context.actorOf(props, name = actorName)
     } else {
       ??? //TODO: Implement!
     }
+
+  private def fetchCallInputEntries(callKey: JobKey, symbolCache: SymbolCache): Traversable[SymbolStoreEntry] = {
+    symbolCache.getOrElse(SymbolCacheKey(callKey.scope.fullyQualifiedName, input = true), Seq.empty)
+  }
+
+  private def buildSymbolStoreEntries: Traversable[SymbolStoreEntry] = {
+    val actualInputs = workflowDescriptor.backendDescriptor.inputs ++ workflowDescriptor.declarations
+    val inputSymbols = actualInputs map {
+      case (name, value) => SymbolStoreEntry(name, value, None, input = true)
+    }
+
+    val callSymbols = for {
+      call <- workflowDescriptor.namespace.workflow.calls
+      (k, v) <- call.inputMappings
+    } yield SymbolStoreEntry(s"${call.fullyQualifiedName}.$k", v, None, input = true)
+
+    inputSymbols.toSet ++ callSymbols.toSet
+  }
+
+  /**
+    * Attempt to start all runnable calls and return updated state data.  This will create a new copy
+    * of the state data including new pending persists.
+    */
+  @tailrec
+  private def startRunnableCalls(data: WorkflowExecutionActorData): WorkflowExecutionActorData = {
```

Comment:
Personally I really dislike having a function that's almost 1/4 of the entire file, and which contains a whole bunch of sub-functions.

Would it be possible to extract these sub-functions into separate methods so that the flow of `startRunnableCalls` is more obvious?